processor/src/main/kotlin/permissions/dispatcher/processor/impl/java/JavaBaseProcessorUnit.kt

```diff
@@ -43,17 +43,26 @@ abstract class JavaBaseProcessorUnit(val messager: Messager) : JavaProcessorUnit
 
     abstract fun getActivityName(targetParam: String): String
 
+    abstract fun isDeprecated(): Boolean
+
     /* Begin private */
 
     private fun createTypeSpec(rpe: RuntimePermissionsElement, requestCodeProvider: RequestCodeProvider): TypeSpec {
-        return TypeSpec.classBuilder(rpe.generatedClassName)
+        val typeSpecBuilder = TypeSpec.classBuilder(rpe.generatedClassName)
```

Comment:
We can avoid the variable assignment by placing the `isDeprecated()` conditional inside an `apply` block:

```kotlin
return TypeSpec.classBuilder(rpe.generatedClassName)
    // ...
    .apply {
        if (isDeprecated()) {
            addAnnotation(createDeprecatedAnnotation())
        }
    }
    .build()
```

Reply:
Wow nice!